include/podio/detail/OrderKey.h

```diff
@@ -0,0 +1,28 @@
+#ifndef PODIO_DETAIL_ORDERKEY_H
+#define PODIO_DETAIL_ORDERKEY_H
+#include <functional>
+namespace podio::detail {
+// Internal class allowing datatype objects to be placed in data structures like maps and sets by providing a way to
+// compare them. The comparison is based on addresses of their internal data objects.
+//
+// This class is intended to be used as the return value of internal `podio::detail::getOrderKey` free functions. These
+// functions are friends of each datatype, allowing them to access the internal data objects and define less-than
+// comparison operators for both datatypes and interface types.
+//
+// The friend free function design is used in order to reduce the coupling between interfaces and datatypes. Interfaces
+// do not need to be friends of datatypes to define the less-than comparison operator, which allows using datatypes from
+// different datamodels in an interface type.
```

Comment:
```suggestion
/// Internal class allowing datatype objects to be placed in data structures like maps and sets by providing a way to
/// compare them. The comparison is based on addresses of their internal data objects.
///
/// This class is intended to be used as the return value of internal `podio::detail::getOrderKey` free functions. These
/// functions are friends of each datatype, allowing them to access the internal data objects and define less-than
/// comparison operators for both datatypes and interface types.
///
/// The friend free function design is used in order to reduce the coupling between interfaces and datatypes. Interfaces
/// do not need to be friends of datatypes to define the less-than comparison operator, which allows using datatypes from
/// different datamodels in an interface type.
```
Just to be consistent with other docstrings